PolyEngine/Engine/Src/SoundResource.cpp

```diff
@@ -0,0 +1,253 @@
+#include "EnginePCH.hpp"
+
+#include <al.h>
+#include <ogg/ogg.h>
+#include <vorbis\vorbisfile.h>
+#include <fstream>
+
+#include "SoundResource.hpp"
+
+using namespace Poly;
+
+SoundResource::SoundResource(const String& path)
```

Comment:
This whole method seems to be overcomplicated. See this: https://www.gamedev.net/articles/programming/general-and-gameplay-programming/introduction-to-ogg-vorbis-r2031